third_party/amd/backend/compiler.py

```diff
@@ -266,7 +266,7 @@ def make_ttgir(mod, metadata, options):
             passes.ttgpuir.add_remove_layout_conversions(pm)
         amd.passes.ttgpuir.add_reorder_instructions(pm)
         use_block_pingpong = is_pingpong_schedule_enabled(options.arch)
-        if use_block_pingpong and options.num_stages in [2, 4]:
+        if use_block_pingpong and options.num_stages in [2, 3, 4]:
```

Comment:
Do we only allow pingpong to be enabled for pre-vetted scenarios where we know that pingpong is faster than non-pingpong? Or do we allow the user to enable pingpong even when it's slower (e.g. for testing or analysis...)?
I ask because I'm wondering if we distinguish between "supported" cases being numerically correct or performant also.